mne/io/edf/edf.py

```diff
@@ -474,31 +318,24 @@ def _get_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
         warn('Physical range is not defined in following channels:\n' +
              ', '.join(ch_names[i] for i in bad_idx))
         physical_ranges[bad_idx] = 1
-    if 'stim_data' in edf_info and stim_channel == 'auto':  # For GDF events.
+    if 'stim_data' in edf_info:  # For GDF events.  # XXX: com back to this
         cals = np.append(cals, 1)
-    if stim_channel is not None:
-        stim_channel = _check_stim_channel(stim_channel, ch_names, sel)
+    stim_channel = _check_stim_channel(stim_channel, ch_names, sel)
```

Comment:
All this could be change by just chekcing if any of the valid stim keywords is present in ch_names